src/engine/source/indexerconnector/test/unit/monitoring_test.cpp

```diff
@@ -1,169 +1,154 @@
 /*
  * Wazuh Indexer Connector - Monitoring tests
  * Copyright (C) 2015, Wazuh Inc.
- * September 08, 2023.
+ * August 30, 2024.
  *
  * This program is free software; you can redistribute it
  * and/or modify it under the terms of the GNU General Public
  * License (version 2) as published by the FSF - Free Software
  * Foundation.
  */
 
-#include "base/logging.hpp"
-#include "fakeOpenSearchServer.hpp"
-#include "monitoring.hpp"
-#include <chrono>
-#include <gtest/gtest.h>
-#include <memory>
-#include <thread>
+#include "monitoring_test.hpp"
+#include "mockHTTTPRequest.hpp"
+#include "trampolineHTTPRequest.hpp"
 
-/**
- * @brief Runs unit tests for Monitoring class
- */
-class MonitoringTest : public ::testing::Test
+auto mockHTTPRequestLambda =
+    [](const std::string& greenResponse, const std::string& redResponse, const std::string& yellowResponse)
 {
-protected:
-    inline static std::unique_ptr<FakeOpenSearchServer>
-        m_fakeOpenSearchGreenServer; ///< pointer to FakeOpenSearchServer class
-
-    inline static std::unique_ptr<FakeOpenSearchServer>
-        m_fakeOpenSearchRedServer; ///< pointer to FakeOpenSearchServer class
-
-    std::shared_ptr<Monitoring> m_monitoring; ///< pointer to Monitoring class
-
-    std::vector<std::string> m_servers; ///< Servers
-
-    /**
-     * @brief Sets initial conditions for each test case.
-     */
-    // cppcheck-suppress unusedFunction
-    void SetUp() override
-    {
-        logging::testInit();
-        // Register the host and port of the green server
-        m_servers.emplace_back("http://localhost:9209");
-        // Register the host and port of the red server
-        m_servers.emplace_back("http://localhost:9210");
-    }
-
-    /**
-     * @brief Creates the fakeOpenSearchServers for the runtime of the test suite
-     */
-    // cppcheck-suppress unusedFunction
-    static void SetUpTestSuite()
+    return [greenResponse, redResponse, yellowResponse](RequestParameters requestParameters,
+                                                        PostRequestParameters postRequestParameters,
+                                                        ConfigurationParameters /*configurationParameters*/)
     {
-        const std::string host {"localhost"};
-
-        if (!m_fakeOpenSearchGreenServer)
+        const auto& url = requestParameters.url.url();
+        if (url == GREEN_SERVER + "/_cat/health?v")
         {
-            m_fakeOpenSearchGreenServer = std::make_unique<FakeOpenSearchServer>(host, 9209, "green");
+            postRequestParameters.onSuccess(greenResponse);
         }
-
-        if (!m_fakeOpenSearchRedServer)
+        else if (url == RED_SERVER + "/_cat/health?v" || url == YELLOW_SERVER + "/_cat/health?v")
```

Comment:
This was clarified in today's dayli, I believe it is not due to a lack of personal attention since before implementing this change I consulted you if the logic was the same and proceeded once you confirmed it (Ref: https://github.com/wazuh/wazuh/pull/25539#discussion_r1742844708).

Additionally, the logic implemented in `monitoring. hpp` currently only checks if the state is green, treating the other states similarly, and this is a unit test scenario.

In order to be able to close this task and avoid misunderstandings, the change request to be made is that the message to be sent is different if the server status is yellow or red?

Reply:
Miss communication happens, an improvement for the future 💪🏽